test/barrier/barrier.mli

```diff
@@ -0,0 +1,52 @@
+(** A barrier is a synchronisation tool.
```

Comment:
This is a very minor thing, but in case you are using Emacs, then there is command [`fill-paragraph`](https://www.emacswiki.org/emacs/FillParagraph) that one can use to format paragraphs of text to fill the desired number of columns (i.e. `fill-column`).

Starting from this:

```
    A barrier of capacity [n] blocks domains until [n] of them are
    waiting. Then these [n] domains can pass. Then the barrier is
    reset.

    Note that this barrier is not starvation-free if there is more
    domains trying to pass it than its capacity.

    This module has been written to help make sure that in `qcheck` tests and
    unitary tests, multiple domains are actually running in parallel.
```

having `fill-column` set to 80,  one gets to this:

```
    A barrier of capacity [n] blocks domains until [n] of them are waiting. Then
    these [n] domains can pass. Then the barrier is reset.

    Note that this barrier is not starvation-free if there is more domains
    trying to pass it than its capacity.

    This module has been written to help make sure that in `qcheck` tests and
    unitary tests, multiple domains are actually running in parallel.
```

simply by running `fill-paragraph` (bound to `M-q` key) on each paragraph in a couple of seconds.

Having the text formatted more consistently makes the source code easier to read.  With proper editor support it is also easier to edit (no more randomly adding line breaks, just `M-q` / `fill-paragraph` and you are done).